DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1

```diff
@@ -1,147 +1,168 @@
+    [switch]$Force, 
+
+    # Package and publish the modules listed in $ModuleNameList based on PowerShell module path content.
+    [string[]]$ModuleNameList
+    )
+
+    # Create working directory
+    $TempFolder = "$pwd\temp"
+    New-Item -Path $TempFolder -ItemType Directory -Force -ErrorAction SilentlyContinue
+
+    # Copy the mof documents from the $Source to working dir
+    Copy-Item -Path "$Source\*.mof" -Destination $TempFolder -Force -Verbose
+
+    # Start Deployment!
+    Log -Scope $MyInvocation -Message 'Start Deployment'
+    CreateZipFromPSModulePath -ListModuleNames $ModuleNameList -Destination $TempFolder
+    CreateZipFromSource -Source $Source -Destination $TempFolder
+    # Generate the checkSum file for all the zip and mof files.
+    New-DSCCheckSum -Path $TempFolder -Force
+    # Publish mof and modules to pull server repositories
+    PublishModulesAndChecksum -Source $TempFolder
+    PublishMofDocuments -Source $TempFolder
+    # Deployment is complete!
+    Remove-Item -Path $TempFolder -Recurse -Force -ErrorAction SilentlyContinue
+    Log -Scope $MyInvocation -Message 'End Deployment'
 
 }
 
 #Package the modules using powershell module path
 function CreateZipFromPSModulePath
 {
-   param($listModuleNames, $destination)
+    param($ListModuleNames, $Destination)
+
     # Move all required  modules from powershell module path to a temp folder and package them
-    if(($listModuleNames -eq $null) -or ($listModuleNames.Count -eq 0))
+    if ([string]::IsNullOrEmpty($ListModuleNames))
     {
-        Write-Host "No additional modules are specified to be packaged."
+        Log -Scope $MyInvocation -Message "No additional modules are specified to be packaged." 
     }
-    foreach ($module in $listModuleNames)
+    
+    foreach ($Module in $ListModuleNames)
     {
-        $allVersions = Get-Module -Name $module -ListAvailable -Verbose        
-        #package all versions of the module
-        foreach($moduleVersion in $allVersions)
+        $AllVersions = Get-Module -Name $Module -ListAvailable -Verbose        
+        # Package all versions of the module
+        foreach ($ModuleVersion in $AllVersions)
         {
-            $name = $moduleVersion.Name
-            $source = "$destination\$name"
-            #Create package zip
-            $path  = $moduleVersion.ModuleBase
-            Compress-Archive -Path "$path\*" -DestinationPath "$source.zip" -Verbose -Force 
-            $version = $moduleVersion.Version.ToString()
-            $newName = "$destination\$name" + "_" + "$version" + ".zip"
+            $Name   = $ModuleVersion.Name
+            $Source = "$Destination\$Name"
+            # Create package zip
+            $Path    = $ModuleVersion.ModuleBase
+            $Version = $ModuleVersion.Version.ToString()
+            Log -Scope $MyInvocation -Message "Zipping $Name ($Version)"
+            Compress-Archive -Path "$Path\*" -DestinationPath "$Source.zip" -Verbose -Force 
```

Comment:
This module is neither a dsc resource nor a dsc resource module. It's just coincidentally placed in this github repo because you have write permissions to the repository.